apps/server/src/modules/sharing/sharing.module.ts

```diff
@@ -1,25 +1,26 @@
 import { Module } from '@nestjs/common';
-import { CourseRepo, LessonRepo } from '@shared/repo';
 import { LoggerModule } from '@src/core/logger';
 import { AuthorizationModule } from '@src/modules/authorization';
-import { LearnroomModule } from '@src/modules/learnroom';
+import { CourseRepo } from '@shared/repo';
 import { ShareTokenController } from './controller/share-token.controller';
 import { ShareTokenUC } from './uc';
 import { ShareTokenService, TokenGenerator } from './service';
 import { ShareTokenRepo } from './repo/share-token.repo';
+import { CourseService } from '../learnroom/service/course.service';
 import { LessonModule } from '../lesson';
+import { LearnroomModule } from '../learnroom';
 
 @Module({
-	imports: [AuthorizationModule, LearnroomModule, LessonModule, LoggerModule],
+	imports: [AuthorizationModule, LoggerModule, LessonModule],
 	controllers: [],
-	providers: [ShareTokenService, TokenGenerator, ShareTokenRepo, CourseRepo, LessonRepo],
+	providers: [ShareTokenService, TokenGenerator, ShareTokenRepo, CourseService, CourseRepo],
```

Comment:
Can't import CourseService without CourseRepo

Reply:
when you import the learnroom Module (like you do the LessonModule), it will provide the CourseService, so there is no need to provide it again yourself.

Reply:
Fixed, just had to add CourseService to learnroom.module.ts
https://github.com/hpi-schul-cloud/schulcloud-server/blob/c31ad896c0164011be6fa3cdacd98112e402ad56/apps/server/src/modules/learnroom/learnroom.module.ts#L71